src/sync-pending-events.ts

```diff
@@ -129,20 +132,16 @@ export class EventTracker {
     db: EventDBHelper
   ) {
     let lastCheckedBlockNumber = await db.getLastMonitorBlock();
-    let latestBlockNumber = await getLatestBlockNumber(this.source);
-    let trueLatestBlockNumber = await getValidBlockNumber(this.source, lastCheckedBlockNumber, latestBlockNumber);
-    if (trueLatestBlockNumber) {
-      latestBlockNumber = trueLatestBlockNumber;
-    }else {
-      latestBlockNumber = lastCheckedBlockNumber;
-    }
     if(lastCheckedBlockNumber < this.eventSyncStartingPoint) {
       lastCheckedBlockNumber = this.eventSyncStartingPoint;
       console.log("Chain Height Before Deployment: " + lastCheckedBlockNumber + " Is Used");
     }
+    let latestBlockNumber = await getLatestBlockNumberFromSource(this.source);
+    let trueLatestBlockNumber = await getValidBlockNumber(this.source, lastCheckedBlockNumber, latestBlockNumber);
```

Comment:
right is getValidBlockNumber(), and left is trueLatestBlockNumber. Better use the same name convention, either "valid" or "trueLatest", otherwise it is very confusing what is the meaning of getValidBlockNumber. (In this case, I guess the name of getValidBlockNumber is misleading)

Reply:
Need change and make it consistent